prov/gni/src/gnix_rma.c

```diff
@@ -1352,8 +1352,10 @@ ssize_t _gnix_rma(struct gnix_fid_ep *ep, enum gnix_fab_req_type fr_type,
 	}
 
 	/* Adding FI_FENCE to an FI_MORE list will break the FI_MORE Chain.
-	 * Remove FI_MORE flag when FI_FENCE flag is present. */
-	if (flags & FI_FENCE) {
+	 * Current FI_MORE implementation does not create remote CQ events.
+	 * Remove FI_MORE flag when FI_FENCE or FI_REMOTE_CQ_DATA flags are
+	 * present. */
+	if ((flags & FI_FENCE) || (flags & FI_REMOTE_CQ_DATA)) {
```

Comment:
I think you also need to check for:

req->vc->peer_caps & FI_RMA_EVENT

In this case, we send an event to the target to increment a remote counter.  We don't support that with FI_MORE.

Once that's done you should be able to remove the 'fi_more_set' workaround from our criterion tests.